crates/goose/src/providers/openai.rs

```diff
@@ -24,14 +24,20 @@ pub struct OpenAiProvider {
     model: ModelConfig,
 }
 
-impl OpenAiProvider {
-    pub fn from_env() -> Result<Self> {
-        let api_key = crate::key_manager::get_keyring_secret("OPENAI_API_KEY", Default::default())?;
-        let host =
-            std::env::var("OPENAI_HOST").unwrap_or_else(|_| "https://api.openai.com".to_string());
-        let model_name =
-            std::env::var("OPENAI_MODEL").unwrap_or_else(|_| OPEN_AI_DEFAULT_MODEL.to_string());
+impl Default for OpenAiProvider {
+    fn default() -> Self {
```

Comment:
looks like these methods are only used in tests - is that correct? i was a bit confused cause we have `default_model` on provider metdata and then, the `model_name` in  ModelConfig

Reply:
yes that's right, the app was sending model but it was not getting used, so i made the normal usage path more explicit. Defaults right now are for scripting/tests/etc